RDKB-60970: RDKLogger OpenSource Migration

dshett549 opened the pull request:
Reason for change: To have single version of rdklogger across RDKB, RDKV, RDKC & RDKE 
Test Procedure: Tested and verified 
Risks:Medium 
Priority:P1

dshett549 committed:
RDKB-60970: RDKLogger OpenSource Migration

Reason for change: To fix compilation errors
Test Procedure: Tested and verified
Risks:Medium
Priority:P1

Signed-off-by: dshett549 <DEEPTHICHANDRASHEKAR_SHETTY@comcast.com>

dshett549 committed:
Update rdk_debug.h

dshett549 committed:
Update rdk_debug.h

dshett549 committed:
Remove onboard_log macros

Signed-off-by: dshett549 <DEEPTHICHANDRASHEKAR_SHETTY@comcast.com>

dshett549 committed:
To link log4c

Signed-off-by: dshett549 <DEEPTHICHANDRASHEKAR_SHETTY@comcast.com>

dshett549 committed:
To resolve compilation errors

Signed-off-by: dshett549 <DEEPTHICHANDRASHEKAR_SHETTY@comcast.com>

dshett549 committed:
To link log4c

Signed-off-by: dshett549 <DEEPTHICHANDRASHEKAR_SHETTY@comcast.com>

dshett549 committed:
To resolve workflow error

Signed-off-by: dshett549 <DEEPTHICHANDRASHEKAR_SHETTY@comcast.com>

dshett549 committed:
Remove Macros

Signed-off-by: dshett549 <DEEPTHICHANDRASHEKAR_SHETTY@comcast.com>

dshett549 committed:
Link log4c library

Signed-off-by: dshett549 <DEEPTHICHANDRASHEKAR_SHETTY@comcast.com>

dshett549 committed:
Update Makefile.am

Signed-off-by: dshett549 <DEEPTHICHANDRASHEKAR_SHETTY@comcast.com>

rdkcmf-jenkins commented:
b'## **Blackduck scan failure details**

**Summary**: 0 **violation**s, 0 files pending approval, 1 file pending identification.
* Protex Server Path: /home/blackduck/github/rdk_logger/20/rdk/components/generic/rdk_logger

* Commit: 70880b09b2509e808365ac8ae931388bf224a5e0

 **Report detail:** [gist](https://gist.github.com/rdkcmf-jenkins/f8e0f18a6e3fad628ce9bde1956e88b3)'

dshett549 committed:
Update Makefile.am

Signed-off-by: dshett549 <DEEPTHICHANDRASHEKAR_SHETTY@comcast.com>

rdkcmf-jenkins commented:
b'## Copyright scan failure
 **Commit:** dc0159e37219f58ca8cd03d5c566e23f9d7cb870
**Report detail:**  https://gist.github.com/rdkcmf-jenkins/0936e88a797f9b66d1c4f5cd5d65dfc6'

dshett549 committed:
Updated NOTICE, LICENSE and Makefile.am

Signed-off-by: dshett549 <DEEPTHICHANDRASHEKAR_SHETTY@comcast.com>

karuna2git committed:
Updated logger directory

1. Updated the workflow & cleaned up the Makefiles
2. Moved rdk_debug.h as rdk_logger.h and added temporary rdk_debug.h which just includes the primary one.
3. Cleaned up the code to use new rdk_logger.h
4. Added coverity shell scripts

karuna2git committed:
Added Legacy rdk_dbg_MsgRaw() to support pre-compiled soc library that uses legacy rdklogger

karuna2git committed:
Fixed the error in Camera devices specific build error

Signed-off-by: kamirt573_comcast <karunakaran_amirthalingam@cable.comcast.com>

karuna2git commented:
Hi @mhughesacn ,
Regarding rdk_logger.h
`
 Error: Source code with 'GPL-2.0' license found in local file 'include/rdk_logger.h' (Match: gnuton/asuswrt-merlin.ng/3006.102.1.1-gnuton0.beta2, 15 lines, url: https://github.com/gnuton/asuswrt-merlin.ng/archive/refs/tags/3006.102.1_1-gnuton0_beta2.tar.gz, file: release/src-rt-5.04behnd.4916/userspace/public/include/wifi_hal.h)
`
It is RDK licensed code that is present in `asuswrt-merlin.ng` Ref: https://github.com/gnuton/asuswrt-merlin.ng/blob/e39722f750ec7773ecd75eb9115c681ea6b5238e/release/src-rt-5.04behnd.4916/userspace/public/include/wifi_hal.h

Regarding onboard_main.c,
`
Error: Source code with 'Apache-2.0' license found in local file 'utils/rdk_logger_onboard_main.c' (Match: rdk/components/generic/sys_mon_tools/sys_resource/rdk/components/generic/sys_mon_tools/sys_resource/1, 18 lines, url: https://code.rdkcentral.com/r/plugins/gitiles/rdk/components/generic/sys_mon_tools/sys_resource/+archive/RDKB-TEST-RELEASE-1.tar.gz, file: src/debugRecord.cpp)
`
It is RDK code hosted by RDK team; We can remove the sys_mon_tools.

Thanks,
Karunakaran A
